File: Algorithms/C++/204-Count_Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n <= 0) return 0;
        vector<bool> prime(n, true);
        prime[0] = false;
        prime[1] = false;
        for (int i = 0; i < sqrt(n); ++i) {
            if (prime[i]) {
                for (int j = i * i; j < n; j += i) {
                    prime[j] = false;
                }
            }
        }
        return count(prime.begin(), prime.end(), true);
    }
};

File: Algorithms/Python3.x/204-Count_Primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 3:
            return 0
        primes = [True] * n
        primes[0], primes[1] = False, False
        for i in range(2, int(n ** 0.5) + 1):
            if primes[i]:
                primes[i * i:n:i] = [False] * len(primes[i * i:n:i])
        return sum(primes)